### DISCUSSION TITLE
BM8563

### DISCUSSION THREAD

**Discussion by ririfonfon** [Makuna/Rtc#188] [General]
Good morning,
would it be possible to add the rtc BM8563 to your superb library.
indeed I developed a code with a ds3231 and I would like to change this one by M5Stamp-TimerPower
@https://github.com/m5stack/M5Stamp-TimerPower/tree/main
@https://github.com/m5stack/M5Stamp-TimerPower/blob/main/FacTest/M5_I2C_BM8563.h
@https://github.com/m5stack/M5Stamp-TimerPower/blob/main/FacTest/M5_I2C_BM8563.cpp
I found this, if it helps
@https://github.com/AdalbertSZ/BM8563/blob/main/BM8563.h
@https://github.com/AdalbertSZ/BM8563/blob/main/BM8563.cpp
THANKS

**Comment by Makuna** [comment]
Where did you get your module?  I just checked around and could not find a module that uses this chip.  
Once I can get one, I can add support.

**Comment by ririfonfon** [comment]
M5Stamp Timer Power (BM8563)shop.m5stack.comHereRichard FontaineEnvoyé de mon iPadLe 20 juil. 2023 à 17:00, Michael Miller ***@***.***> a écrit :﻿
Where did you get your module?  I just checked around and could not find a module that uses this chip.
Once I can get one, I can add support.

**Reply by Makuna** [reply]
That seems to be a very specific designed power module that happens to include BM8563 for wake up ability.  I can't test the BM8563 with that module as it doesn't expose all the pins.

**Comment by ririfonfon** [comment]
Thanks for trying!

Le ven. 21 juil. 2023 à 17:38, Michael Miller ***@***.***> a
écrit :

**Comment by Makuna** [comment]
I noticed a PCF8563, I wonder if they are clones of each other?

**Reply by Makuna** [reply]
After reviewing the spec sheets, they ARE clones of each other.

**Comment by Makuna** [comment]
https://github.com/Makuna/Rtc/issues/189

**Comment by Makuna** [comment]
Something I noticed after reviewing the spec sheet, is that these do not have battery backup time unless they always run on a battery.  They have no circuits to either charge or switch to a circuit supplied battery when the main power is removed.  They either run on battery all the time or run on VCC all the time.  This seems rather poor feature for a RTC where one of its benefits is once set they retain a running clock even when power is lost.  

It's just something to keep in mind when choosing this RTC.

All the DS#### RTCs support this backup battery feature.

**Comment by Makuna** [comment]
@ririfonfon I have a PCF8563 on order.  I will have a branch with the support before it arrives if you would like to try it out.  See https://github.com/Makuna/Rtc/tree/PCF8563.  You can clone this into the same directory the library is already present on your machine and it will just work.

**Comment by ririfonfon** [comment]
Great thank you.I'm dying to try this.Richard FontaineEnvoyé de mon iPadLe 23 juil. 2023 à 01:23, Michael Miller ***@***.***> a écrit :﻿
@ririfonfon I have a PCF8563 on order.  I will have a branch with the support before it arrives if you would like to try it out.  Let me know and I will publish the branch so you can give it a try.
